File: implementations/csharp/src/RCL.cs
using System.Collections.Generic;

namespace RCLImpl {
public static class RCL {
  public static DocumentNode Parse(string s){ return Core.RCL.Parse(s); }
  public static string Format(string s){ return Formatter.Format(Parse(s)); }
  public static string Format(DocumentNode d){ return Formatter.Format(d); }
  public static Dictionary<string, object> ToObject(string s){ return Core.RCL.ToObject(s); }
  public static Dictionary<string, object> ToObject(DocumentNode d){ return Core.RCL.ToObject(d); }
  public static string ToYAML(string s){ return Converters.ToYAML(Parse(s)); }
  public static string ToYAML(DocumentNode d){ return Converters.ToYAML(d); }
  public static string ToTOML(string s){ return Converters.ToTOML(Parse(s)); }
  public static string ToTOML(DocumentNode d){ return Converters.ToTOML(d); }
  public static string ToHCL(string s){ return Converters.ToHCL(Parse(s)); }
  public static string ToHCL(DocumentNode d){ return Converters.ToHCL(d); }
}
}

File: implementations/ruby/lib/rcl.rb
require_relative "rcl/parser"
require_relative "rcl/formatter"
require_relative "rcl/converters"

module RCL
  module_function

  def parse(text)
    Parser.new(text).parse
  end

  def format(ast)
    Formatter.new.format(ast)
  end

  def to_object(ast)
    Converters.to_object(ast)
  end

  def to_yaml(ast)
    Converters.to_yaml(ast)
  end

  def to_toml(ast)
    Converters.to_toml(ast)
  end

  def to_hcl(ast)
    Converters.to_hcl(ast)
  end
end
